katz/src/main/kotlin/katz/Id.kt

```diff
@@ -0,0 +1,31 @@
+package katz
+
+class Id<out A>(val value: A): HK<Id.F, A> {
+
+    class F
+
+    inline fun <B> map(f: (A) -> B): Id<B> = Id(f(value))
+
+    inline fun <B> flatMap(f: (A) -> Id<B>): Id<B> = f(value)
+
+    companion object : IdInstances
+}
+
+/**
+ * Companion interface for [Id] type
+ * outside due to https://youtrack.jetbrains.com/issue/KT-10532
+ */
+private interface IdInstances {
+    fun <A> HK<Id.F, A>.ev(): Id<A> = this as Id<A>
+
+    fun monad() = object : Monad<Id.F> {
+        override fun <A, B> map(fa: HK<Id.F, A>, f: (A) -> B): HK<Id.F, B> =
```

Comment:
Will `map` and `flatMap` methods always have the same form? In this case and Option the answers is yes, it's possible extract it to parent interface?